client/src/components/AuthRoute/AuthRoute.tsx

```diff
@@ -0,0 +1,15 @@
+import { Redirect, Route, RouteProps } from 'react-router-dom';
+import { useAuth } from '../../context/useAuthContext';
+import { CircularProgress } from '@mui/material';
+import React from 'react';
+
+export const AuthRoute = (props: RouteProps) => {
+  const { loggedInUser } = useAuth();
+  return loggedInUser === undefined ? (
```

Comment:
Good work! I wouldn't consider using a ternary operator for nested if/else clauses (to increase readability and maintainability)

```
if (!loggedInUser) {
    return <CircularProgress />
}

return loggedInUser ? <Route {...props}> : <Redirect to='/login' />
```